File: src/main/java/com/itechart/demojavakotlin/controller/UserController.java
package com.itechart.demojavakotlin.controller;

import com.itechart.demojavakotlin.entity.UserEntity;
import com.itechart.demojavakotlin.model.UserRequest;
import com.itechart.demojavakotlin.service.UserService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
@RequestMapping("/users")
public class UserController {

    private final UserService userService;

    @PostMapping
    public ResponseEntity<UserEntity> create(@RequestBody final UserRequest userRequest) {
        return new ResponseEntity<>(userService.createUser(userRequest), HttpStatus.OK);
    }
}

File: kotlinpath/demojavakotlin/src/main/kotlin/by/itechart/demojavakotlin/controller/UserController.kt
package by.itechart.demojavakotlin.controller

import by.itechart.demojavakotlin.entity.UserEntity
import by.itechart.demojavakotlin.model.UserRequest
import by.itechart.demojavakotlin.service.UserService
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/users")
class UserController(private val userService: UserService) {

    @PostMapping
    fun create(@RequestBody userRequest: UserRequest): ResponseEntity<UserEntity> {
        return ResponseEntity<UserEntity>(userService.createUser(userRequest), HttpStatus.OK)
    }

}
